src/spyglass/utils/dj_merge_tables.py

```diff
@@ -805,22 +821,33 @@ def _master_table_pairs(
     conn = connection or table_list[0].connection
 
     master_table_pairs = []
+    unique_parts = []
+
     # Adapted from Spyglass PR 535
     for table in table_list:
-        master_name = get_master(table.full_table_name)
+        table_name = table.full_table_name
+        if table_name in unique_parts:  # then repeat in list
+            continue
+
+        master_name = get_master(table_name)
         if not master_name:  # then it's not a part table
             continue
 
         master = dj.FreeTable(conn, master_name)
-
         if RESERVED_PRIMARY_KEY not in master.heading.attributes.keys():
-            continue
-
-        restricted_table = table.restrict(restriction)
+            continue  # then it's not a merge table
 
-        if not restricted_table:
+        restricted_join = restricted_parent * table
```

Comment:
This is definitely safer than before. Might still be tedious to figure in cases where the parts table no longer shares keys with the restricted_parent (thinking of CurationV1 and Session)

Reply:
Thanks for reviewing. After talking to Eric, we decided to split out the admin/delete pieces, but I'll certainly fix the array error for next time. I can try out the CurationV1 case. Note sure how else I would approach though. Would be interested in any alternative that comes to mind for you

Reply:
I tested this case with the new spike sort merge table, and it successfully captured using a valid restriction for the upstream table
![image](https://github.com/LorenFrankLab/spyglass/assets/9404707/48d8ab7b-ca65-4200-bfb6-61371b61781e)

Reply:
Oh, awesome! I think I just misunderstood what table was being joined together for the restrictiction